src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -457,7 +458,18 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
-func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
+func prepareAdditionalArguments(
+	developerMode bool, limits, requests corev1.ResourceList,
+) []string {
+	var smp int64 = 1
+	if !limits.Cpu().IsZero() {
+		limits.Cpu().RoundUp(0)
```

Comment:
I wonder if the CPU request is non-integer, or, at least, below 1, we should run overprovisioned?

Reply:
We do that for developer mode. Are you thinking of doing so for both modes?

Reply:
I don't know. If non-developerMode works fine with TLS then all good. I never got to the bottom of it, but when I was testing I recall that we added --reserve-memory when rpk was taught to pass in TLS. My guess what that it consumes some memory before forking. I think we tested with 1 core and 1GiB - 1GiB being the minimum per core (2GiB is recommended), and it failed.

Reply:
E.g., if CPU request is set to 200m in non-developerMode, then overprovisioned might help it play nicely with other processes on the machine. It's not a recommended config. I don't think this is a showstopper.